persistence/persistence-jdbc/src/main/java/uk/gov/justice/services/jdbc/persistence/AbstractJdbcRepository.java

```diff
@@ -42,32 +43,51 @@ protected DataSource getDataSource() {
         return datasource;
     }
 
-    protected PreparedStatementWrapper preparedStatementWrapperOf(final String selectByStreamId) throws SQLException {
-        return valueOf(getDataSource().getConnection(), selectByStreamId);
+    protected PreparedStatementWrapper preparedStatementWrapperOf(final String query) throws SQLException {
+        return valueOf(getDataSource().getConnection(), query);
     }
 
     protected Stream<T> streamOf(final PreparedStatementWrapper psWrapper) throws SQLException {
 
         final ResultSet resultSet = psWrapper.executeQuery();
 
-        return StreamSupport.stream(new Spliterators.AbstractSpliterator<T>(
+        return streamOf(psWrapper, resultSet);
+    }
+
+    protected <U> Stream<U> streamOf(final PreparedStatementWrapper psWrapper, final ResultSet resultSet, final Function<ResultSet, U> rs2EntityFunction) {
```

Comment:
Is there a better name for this ? rs2EntityFunction
Is it resultSetEntityMapperFunction or something?

Might be useful to have a javadoc comment, explaining what U is etc.